### ISSUE TITLE
can ein work with jupyterlab-lsp?

### ISSUE THREAD

**Issue by lyjdwh** [millejoh/emacs-ipython-notebook#755]
I notice https://github.com/krassowski/jupyterlab-lsp provides really good lsp support!

Can jupyterlab-lsp work with ein?

**Comment by dickmao** [comment]
No.  EIN is ostensibly about getting real work done.  Jupyterlab plugins generally add whizbang cosmetics which, arguably, engender a false sense of productivity.  Moreover, the plugin ecosystem is a typescript, web2.0 technology of which emacs is hopelessly ignorant.  Thanks for your interest.

**Comment by eeshugerman** [comment]
It sounds like OP is interested in LSP features in EIN. @dickmao, you consider LSP features to be whizbang cosmetics?

Note that the project linked includes a server-side component: [jupyter-lsp on PyPI](https://pypi.org/project/jupyter-lsp/). I imagine lsp-mode would first need to add support for jupyter-lsp, but if that were to happen, could you tell us what else it would take to get it working with EIN? Would changes to EIN even be necessary?

**Comment by dickmao** [comment]
I'm afraid we're not on the same page. It takes five minutes to speculate about what's possible.  But getting a reasonable estimate of what would be required takes *at least* an hour.  I spent fifteen minutes just now, so I still don't know what's required, but the number of moving parts, whether the server-side python needs to be mated with the typescript, and the fragility of EIN and lsp-mode put me off it pretty quick.

The first step is just getting barebones lsp-mode to work in EIN.  Last I checked it doesn't work.
